### PR TITLE
Remove some dead remoting and context static things

### PR DESCRIPTION
**Pull request by AustinWise**
Building on #20256, this further cleans up references to context statics. It also removes some adjacent remoting things.

### CODE REVIEW

**Comment by AustinWise** [comment]
One thing I'm not sure is ok is renaming `DacDbiInterfaceImpl::GetThreadOrContextStaticAddress` to `DacDbiInterfaceImpl::GetThreadStaticAddress`. I don't know if this is a stable interface or not.

**Comment by janvorli** [comment]
@dotnet-bot test Ubuntu x64 Checked Innerloop Build and Test please

**Comment by janvorli** [comment]
> One thing I'm not sure is ok is renaming DacDbiInterfaceImpl::GetThreadOrContextStaticAddress to DacDbiInterfaceImpl::GetThreadStaticAddress. I don't know if this is a stable interface or not.

I think the renaming is ok. @mikem8361 am I right that there is no external dependency on the name itself?

File: src/debug/daccess/dacdbiimpl.cpp [hunk 1]
```diff
@@ -3217,20 +3217,15 @@ TypeHandle DacDbiInterfaceImpl::ExpandedTypeInfoToTypeHandle(DebuggerIPCE_Expand
...
-// If this is a context-static, the function uses the thread's current context.  
-// [This is important because it means that you can't lookup a context static 
-// unless you have a thread in that context. If anybody actually cared about contexts, 
-// we might have to revise this in the future] 
+// The address is constant and could be cached.
 // 
 // This can commonly fail, in which case, it will return NULL.
 // ----------------------------------------------------------------------------
-CORDB_ADDRESS DacDbiInterfaceImpl::GetThreadOrContextStaticAddress(VMPTR_FieldDesc vmField,
+CORDB_ADDRESS DacDbiInterfaceImpl::GetThreadStaticAddress(VMPTR_FieldDesc vmField,
```
Comment (jkotas) [inline src/debug/daccess/dacdbiimpl.cpp]: Nit: Could you please fix the indentation of the second argument and in other places where GetThreadOrContextStaticAddress was renamed to GetThreadStaticAddress?

**Comment by jkotas** [comment]
> renaming `DacDbiInterfaceImpl::GetThreadOrContextStaticAddress` to `DacDbiInterfaceImpl::GetThreadStaticAddress`

Yes, this rename should be ok.

File: src/debug/daccess/dacdbiimpl.cpp (hunk 1 above)
Comment (AustinWise) [reply src/debug/daccess/dacdbiimpl.cpp]: Done.

**Comment by jkotas** [comment]
Thanks!

**Pull request merged**